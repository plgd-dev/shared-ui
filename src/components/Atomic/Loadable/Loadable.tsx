import React, { FC } from 'react'

import IconLoader from '../Loader/IconLoader'
import { Props } from './Loadable.types'

const Loadable: FC<Props> = (props) => {
    const { condition, children } = props
    return condition ? children : <IconLoader center className='loader-icon' size={20} type='secondary' />
}

Loadable.displayName = 'Loadable'

export default Loadable
