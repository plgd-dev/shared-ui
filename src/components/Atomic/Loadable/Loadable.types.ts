import { ReactElement } from 'react'

export type Props = {
    condition: boolean
    children: ReactElement
}
