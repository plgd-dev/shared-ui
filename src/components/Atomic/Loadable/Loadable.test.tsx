import { render } from '../../../../test/jest-wrapper'
import Loadable from './Loadable'

describe('<Loadable>', () => {
    it('renders children when condition is true', () => {
        const { container, asFragment } = render(
            <Loadable condition={true}>
                <div>Test</div>
            </Loadable>
        )
        expect(asFragment()).toMatchSnapshot()
        expect(container).toHaveTextContent('Test')
    })

    it('renders IconLoader when condition is false', () => {
        const { container } = render(
            <Loadable condition={false}>
                <div>Test</div>
            </Loadable>
        )

        const iconLoader = container.querySelector('.loader-icon')
        expect(iconLoader).toBeInTheDocument()
    })
})
