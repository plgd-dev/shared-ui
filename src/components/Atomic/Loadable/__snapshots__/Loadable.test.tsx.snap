// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loadable> renders children when condition is true 1`] = `
<DocumentFragment>
  <div>
    Test
  </div>
</DocumentFragment>
`;
